### PR TITLE
feat: integer_casting

### PR DESCRIPTION
**Pull request by hemantjoestar**
An attempt at what was suggested by
https://twitter.com/PapiniShahar/status/1650939873480548365

- - -
This change is [Reviewable](https://reviewable.io/reviews/starkware-libs/cairo/2938)

### CODE REVIEW

**Review by spapinistarkware — CHANGES_REQUESTED** [review]
> Reviewed 1 of 2 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/cairo/2938)* status: 1 of 2 files reviewed, 1 unresolved discussion (waiting on @hemantjoestar)

___
*[`corelib/src/test/integer_test.cairo` line 684 at r1](https://reviewable.io/reviews/starkware-libs/cairo/2938#-NU3aDlB9tqXpFhbxKCn:-NU3aDlCE-a8foz-DT8s:b-4cs1my) ([raw file](https://github.com/starkware-libs/cairo/blob/c97e98bd9370dd14e0076909ef69d2373a38af24/corelib/src/test/integer_test.cairo#L684)):*
> ```Cairo
> }
> #[test]
> fn should_pass_between_u8_u16() {
> ```

This test can use a single generic function:
```
fn my_test<A, B, impl BIA: BoundedInt<A>, impl BIB: BoundedInt<B>, impl IAB: Into<A,B>,
impl IBA: TryInto<B,A>>{...}
```

And then call while inferering all the impls.

**Comment by hemantjoestar** [comment]
Implemented suggested changes

File: corelib/src/test/integer_test.cairo [hunk 1]
```diff
@@ -678,3 +680,156 @@ fn test_u256_sqrt() {
     let (high, low) = integer::u128_wide_mul(BoundedInt::max(), BoundedInt::max());
     assert(u256_sqrt(as_u256(:high, :low)) == BoundedInt::max(), '(u128::MAX**2)**0.5==u128::MAX');
 }
+
+fn cast_must_pass<A,
```
Comment (hemantjoestar) [inline corelib/src/test/integer_test.cairo:684]: Implemented suggested changes

**Review by spapinistarkware — APPROVED** [review]
:lgtm:

> Reviewed 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/cairo/2938#-:-NUARGnf8zeCirp1I6B9:bnfp4nl)* status: all files reviewed, 1 unresolved discussion (waiting on @hemantjoestar)

**Comment by spapinistarkware** [comment]
Looks good. Please resolve conflicts and make sure cairo-fmt passes.

**Comment by hemantjoestar** [comment]
Before merging ran cairo_fmt.sh locally, but couldn't get the bool line to wrap. `--check` flag too doesn't through any error

**Pull request merged**